packages/core/src/common/command.ts

```diff
@@ -285,24 +286,34 @@ export class CommandRegistry implements CommandService {
      * Execute the active handler for the given command and arguments.
      *
      * Reject if a command cannot be executed.
+     *
+     * **IMPORTANT:** This method has to be synchronous.
      */
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    async executeCommand<T>(commandId: string, ...args: any[]): Promise<T | undefined> {
+    executeCommand<T>(commandId: string, ...args: any[]): Promise<T | undefined> {
         const handler = this.getActiveHandler(commandId, ...args);
-        if (handler) {
-            await this.fireWillExecuteCommand(commandId, args);
-            const result = await handler.execute(...args);
+        if (!handler) {
+            const argsMessage = args && args.length > 0 ? ` (args: ${JSON.stringify(args)})` : '';
```

Comment:
(off-topic) but serializing of random args won't work, just imagine widget 

we have to remove this code

Reply:
as I understand, the args count might be infinite and one of this arg cam represent a widget, am I right?